massa-final-state/src/final_state.rs

```diff
@@ -80,6 +80,20 @@ impl FinalState {
         })
     }
 
+    /// Reset the final state to the initial state.
+    ///
+    /// USED ONLY FOR BOOTSTRAP
```

Comment:
same remark as above: we shouldn't use manual rests here because we will forget one. Writing "only used for bootstrap" is also a sign of bad design. On re-bootstrap we should completely delete all structures and re-create them

Reply:
ok I'll close this discussion in favor of #3698